chapters/command-reference-guide.md

```diff
@@ -0,0 +1,95 @@
+* `ip route delete <ip_prefix_and_subnet>` - delete a route from the routing table
+
+## iptables
+
+[@bork's comic](https://wizardzines.com/comics/iptables/) explaining how to use `iptables`.
+
+`iptables` allows you to write rules for handling incoming, forwarding, or outgoing packets. It can be used as a network layer firewall or just to make simple rules about how to handle traffic on a network.
+
+> Commands used in this guide
+
+* `iptables -L` - show the list of all rules applied to packets processed by the machine
+* `iptables -A OUTPUT -p icmp --icmp-type port-unreachable -j DROP` - generate a new rule to drop any ICMP port unreachable packets
+
+## ping
+
+[@bork's comic](https://wizardzines.com/comics/ping/) explaining `ping` and `traceroute`.
+
+`ping` is a tool that sends ICMP echo request packets to a specified IP address. A machine on that IP address will respond with ICMP echo response packets as long as:
+
+* there is a machine configured on that IP address
+* there are routes from the client issuing the `ping` to and from the machine configured on that IP address
+* the machine configured on that IP address doesn't have configuration settings telling it to ignore ICMP packets
+
+> Example command:
+
+```bash
+ping -c 2 <some_ip_address>
+```
+
+> Helpful flags:
+
+* `-c <some_number>`: Count. Send a specific number of ICMP echo requests. `ping` will wait until its received the ICMP echo response for each request, then exit the program.
+* `-w <some_number>`: Wait. Keep sending ICMP echo requests until the time limit in seconds indicated by the number has passed. `ping` does not wait for the ICMP echo responses.
+
+## tcpdump
+
+[@bork's comic](https://wizardzines.com/comics/tcpdump/) exploring `tcpdump`.
+
+`tcpdump` is a packet sniffing software that allows users to see details about network traffic. It can only listen on a single interface at a time. If you do not indicate an interface specifically, it will listen on whichever interface is defined as its default. To discover what interfaces your machine has, try `ip addr`.
+
+> Example command:
+
+```bash
+tcpdump -ni eth1
+```
+
+> Helpful flags:
+
+* `-e`: Show MAC addresses for each line.
+* `-i <interface_name>`: Select a specific interface.
+* `-n`: Don't convert addresses (e.g., host addresses, port numbers, etc.) to names.
+
+## traceroute
+
+[@bork's comic](https://wizardzines.com/comics/ping/) explaining `ping` and `traceroute`.
+
+`traceroute` is a network analysis tool that shows path a packet takes from the initiator of a request (i.e. a client) to the destination IP. This is done through sending single packets that have a preset `ttl` in the IP header. The `ttl` in IP context does not refer to a specific _time_ to live, instead, it is the maximum number of hops a packet is allowed to traverse before a router should drop it.
+
+> Example command:
+
+```bash
+traceroute 10.1.1.2
+```
```

Comment:
Should we document commands we've never used? Just askin'

Reply:
What do you mean? Do you have suggestions? Or are you saying we shouldn't document `traceroute`? 

https://github.com/psbanka/build-your-own-internet/tree/main/chapters/003-traceroute-aside

Reply:
OH, sorry. I forgot that we actually *did* use traceroute!

Reply:
bahahahaha! I was wondering if that was the case! but I wasn't sure if there was a command that you were thinking of that would be helpful to add? 😆 I'll merge now.